pkg/appmanager/types.go

```diff
@@ -57,52 +61,79 @@ type (
-		SslProfile     *sslProfile     `json:"sslProfile,omitempty"`
-		Policies       []nameRef       `json:"policies,omitempty"`
-		IRules         []string        `json:"rules,omitempty"`
-		// FIXME: All profiles should reside in Profiles, just server ssl ones now.
-		Profiles ProfileRefs `json:"profiles,omitempty"`
+		Balance               string                `json:"balance,omitempty"`
+		Mode                  string                `json:"mode,omitempty"`
+		VirtualAddress        *virtualAddress       `json:"virtualAddress,omitempty"`
+		Destination           string                `json:"destination,omitempty"`
+		Enabled               bool                  `json:"enabled,omitempty"`
+		IpProtocol            string                `json:"ipProtocol,omitempty"`
+		SourceAddrTranslation sourceAddrTranslation `json:"sourceAddressTranslation,omitempty"`
+		SslProfile            *sslProfile           `json:"sslProfile,omitempty"`
+		Policies              []nameRef             `json:"policies,omitempty"`
+		IRules                []string              `json:"rules,omitempty"`
+		Profiles              ProfileRefs           `json:"profiles,omitempty"`
 
 		// iApp parameters
 		IApp                string                    `json:"iapp,omitempty"`
-		IAppPoolMemberTable iappPoolMemberTable       `json:"iappPoolMemberTable,omitempty"`
+		IAppPoolMemberTable *iappPoolMemberTable      `json:"iappPoolMemberTable,omitempty"`
 		IAppOptions         map[string]string         `json:"iappOptions,omitempty"`
 		IAppTables          map[string]iappTableEntry `json:"iappTables,omitempty"`
 		IAppVariables       map[string]string         `json:"iappVariables,omitempty"`
 	}
 
+	// IApp
+	IApp struct {
+		Name                string                    `json:"name"`
+		Partition           string                    `json:"partition,omitempty"`
+		IApp                string                    `json:"template"`
+		IAppPoolMemberTable *iappPoolMemberTable      `json:"poolMemberTable,omitempty"`
+		IAppOptions         map[string]string         `json:"options,omitempty"`
+		IAppTables          map[string]iappTableEntry `json:"tables,omitempty"`
+		IAppVariables       map[string]string         `json:"variables,omitempty"`
+	}
+
+	// Pool Member
+	Member struct {
+		Address			string	`json:"address"`
+		Port			int32	`json:"port"`
+		Session			string	`json:"session,omitempty"`
+	}
+
 	// Pool config
 	Pool struct {
 		Name            string   `json:"name"`
-		Partition       string   `json:"partition"`
+		Partition       string   `json:"partition,omitempty"`
 		Balance         string   `json:"loadBalancingMode"`
-		ServiceName     string   `json:"serviceName"`
-		ServicePort     int32    `json:"servicePort"`
-		PoolMemberAddrs []string `json:"poolMemberAddrs"`
-		MonitorNames    []string `json:"monitor"`
+		ServiceName     string   `json:"serviceName,omitempty"`
+		ServicePort     int32    `json:"servicePort,omitempty"`
+		Members			[]Member `json:"members"`
```

Comment:
This formatting looks messed up. Is "make verify" passing?

Reply:
It's not passing right now, I will make sure it's clean before the next push.